### PR TITLE
Add colorized output for emit

### PR DESCRIPTION
**Pull request by Marwes**
codespan

I haven't spent to much time on this but but I think this should serve as a starting point for #1 (or at least a start/continuation of discussion on it). I would like to get something like this in in any case so that I don't lose out on the source display part in https://github.com/gluon-lang/gluon/pull/505 :)

### CODE REVIEW

**Comment by Marwes** [comment]
Diff https://github.com/brendanzab/codespan/pull/5/files/56d4840a6f9ad770bffb54f56b8ce091c5b6cd1b..fffbb382086c3a97a8030130939fdc5a1d2965bf

File: codespan/src/span.rs [hunk 1]
```diff
@@ -84,11 +84,11 @@ impl<I: Index> Span<I> {
...
     /// let span = Span::new(ByteIndex(3), ByteIndex(6));
-    /// assert_eq!(span.with_lo(ByteIndex(2)), Span::new(ByteIndex(2), ByteIndex(6)));
-    /// assert_eq!(span.with_lo(ByteIndex(5)), Span::new(ByteIndex(5), ByteIndex(6)));
-    /// assert_eq!(span.with_lo(ByteIndex(7)), Span::new(ByteIndex(6), ByteIndex(7)));
+    /// assert_eq!(span.with_start(ByteIndex(2)), Span::new(ByteIndex(2), ByteIndex(6)));
+    /// assert_eq!(span.with_start(ByteIndex(5)), Span::new(ByteIndex(5), ByteIndex(6)));
+    /// assert_eq!(span.with_start(ByteIndex(7)), Span::new(ByteIndex(6), ByteIndex(7)));
     /// ```
-    pub fn with_lo(self, start: I) -> Span<I> {
+    pub fn with_start(self, start: I) -> Span<I> {
```
Comment (brendanzab) [inline codespan/src/span.rs]: 👏

File: codespan-reporting/src/emitter.rs [hunk 2]
```diff
@@ -1,23 +1,92 @@
...
+
 use codespan::CodeMap;
 
+use termcolor::{Color, ColorSpec, WriteColor};
+
 use Diagnostic;
 
-pub fn emit(codemap: &CodeMap, diagnostic: &Diagnostic) {
-    println!("{}: {}", diagnostic.severity, diagnostic.message);
+pub fn emit<W>(mut writer: W, codemap: &CodeMap, diagnostic: &Diagnostic) -> io::Result<()>
```
Comment (brendanzab) [inline codespan-reporting/src/emitter.rs]: Goood. Was wondering if I should switch it to be `eprintln!`, but this is better!

**Comment by brendanzab** [comment]
This is awesome! Super looking forward to it! 😍

File: codespan-reporting/src/emitter.rs (hunk 2 above)
Comment (brendanzab) [reply codespan-reporting/src/emitter.rs]: So is `WriteColor` configurable? ie. my repl has a `--color=[auto|always|never]` flag - would I be able to set that?

Comment (Marwes) [reply codespan-reporting/src/emitter.rs]: Yup, you just set `ColorChoice` when you create the `WriteColor` instance https://github.com/brendanzab/codespan/pull/5/files#diff-a45cab6253488a1ff30335b30b8dc71aR38

Comment (brendanzab) [reply codespan-reporting/src/emitter.rs]: niiiice

**Pull request merged**